src/BuildingBlocks/EventBus/EventBusServiceBus/EventBusServiceBus.cs

```diff
@@ -1,203 +1,191 @@
-                            if (handler == null) continue;
+                        var handler = scope.ResolveOptional(subscription.HandlerType) as IDynamicIntegrationEventHandler;
+                        if (handler == null) continue;
                             
-                            using dynamic eventData = JsonDocument.Parse(message);
-                            await handler.Handle(eventData);
-                        }
-                        else
-                        {
-                            var handler = scope.ResolveOptional(subscription.HandlerType);
-                            if (handler == null) continue;
-                            var eventType = _subsManager.GetEventTypeByName(eventName);
-                            var integrationEvent = JsonSerializer.Deserialize(message, eventType);
-                            var concreteType = typeof(IIntegrationEventHandler<>).MakeGenericType(eventType);
-                            await (Task)concreteType.GetMethod("Handle").Invoke(handler, new object[] { integrationEvent });
-                        }
+                        using dynamic eventData = JsonDocument.Parse(message);
+                        await handler.Handle(eventData);
+                    }
+                    else
+                    {
+                        var handler = scope.ResolveOptional(subscription.HandlerType);
+                        if (handler == null) continue;
+                        var eventType = _subsManager.GetEventTypeByName(eventName);
+                        var integrationEvent = JsonSerializer.Deserialize(message, eventType);
+                        var concreteType = typeof(IIntegrationEventHandler<>).MakeGenericType(eventType);
+                        await (Task)concreteType.GetMethod("Handle").Invoke(handler, new object[] { integrationEvent });
                     }
                 }
-                processed = true;
             }
-            return processed;
+            processed = true;
         }
+        return processed;
+    }
 
-        private void RemoveDefaultRule()
+    private void RemoveDefaultRule()
+    {
+        try
         {
-            try
-            {
-                _serviceBusPersisterConnection
-                    .SubscriptionClient
-                    .RemoveRuleAsync(RuleDescription.DefaultRuleName)
-                    .GetAwaiter()
-                    .GetResult();
-            }
-            catch (MessagingEntityNotFoundException)
-            {
-                _logger.LogWarning("The messaging entity {DefaultRuleName} Could not be found.", RuleDescription.DefaultRuleName);
-            }
+            _serviceBusPersisterConnection
+                .SubscriptionClient
+                .RemoveRuleAsync(RuleDescription.DefaultRuleName)
+                .GetAwaiter()
+                .GetResult();
```

Comment:
```suggestion
                .GetAwaiter()
                .GetResult();
```

This is a code smell.